modules/@apostrophecms/schema/index.js

```diff
@@ -595,6 +588,45 @@ module.exports = {
       name: 'group' // visual grouping only
     });
 
+    self.addFieldType({
+      name: 'range',
+      vueComponent: 'AposInputRange',
+      convert: async function (req, field, data, object) {
+        object[field.name] = self.apos.launder.float(data[field.name], field.def, field.min, field.max);
+        if (field.required && (_.isUndefined(data[field.name]) || !data[field.name].toString().length)) {
+          throw self.apos.error('required');
+        }
+        if (data[field.name] && isNaN(parseFloat(data[field.name]))) {
+          throw self.apos.error('invalid');
+        }
+        // Allow for ranges to go unset
+        // `min` here does not imply requirement, it is the minimum value the range UI will represent
+        if (!data[field.name] || data[field.name] < parseFloat(field.min)) {
```

Comment:
Wouldn't it be invalid if the value was less than the minimum? Is this how we're doing it in 2.x?

Reply:
we are using an out-of-range value as a fake null value, which gets converted here.

There is no UI for the editor to force an out-of-range value so this feels safe to squash and reset back to null, no?

Reply:
Sounds good.

Reply:
i will add a greater-than-max check as well, for completeness